10up/distributor
# Add better support for the block editor's fullscreen mode

### Description of the Change

The default block editor mode is fullscreen, which hides the top admin bar. This admin bar is where Distributor puts the distribution tools. So when in fullscreen mode, it can be hard to figure out how to actually distribute something.

This PR attempts to solve that by adding a new Distributor panel to the editor with a toggle option. This panel will hold the current `Distributed to` and `Distributed on` messages that previously showed in the Post Status panel and will now show a new button to toggle the showing of the admin bar (if needed) and the showing of the Distributor menu content.

This option will show both in normal mode and in fullscreen mode on a published item that isn't a previously Distributed item (an item received from another site). On a non-published item, we output a simple message letting users know they need to publish first. On a newly published item, since the admin bar only updates on page refreshes, instead of showing the toggle we show a message letting users know they need to refresh.

Note: no copy or designs are final. Open to suggestions on all of that.

Screenshots of all scenarios will follow.

### Screenshots
Post received through Distribution, shows the same in both normal and fullscreen mode:
<img width="289" alt="distributed-post-full-and-normal" src="https://user-images.githubusercontent.com/916738/114101444-f9db4780-9882-11eb-8cd2-cd13dab8eb64.png">

Post that has been Distributed, in normal and fullscreen mode
<img width="286" alt="Screen Shot 2021-04-09 at 10 42 40 AM" src="https://user-images.githubusercontent.com/916738/114214371-77559500-9921-11eb-8366-92565ebf20ac.png">

Post with no Distributions, in normal and fullscreen mode
<img width="293" alt="Screen Shot 2021-04-09 at 10 52 14 AM" src="https://user-images.githubusercontent.com/916738/114214567-b257c880-9921-11eb-96c1-7f1487bb6754.png">

Non-published post
<img width="290" alt="non-published-post" src="https://user-images.githubusercontent.com/916738/114101531-19727000-9883-11eb-9e5a-05c049f71d8d.png">

Newly published post, prior to a page refresh
<img width="284" alt="Screen Shot 2021-04-09 at 10 45 56 AM" src="https://user-images.githubusercontent.com/916738/114214641-c7ccf280-9921-11eb-96da-034d5e6e1662.png">

### Alternate Designs

Could attempt to move the full Distribution capabilities from the admin bar to this new panel.

### Benefits

When using the block editor in fullscreen mode, it should now be easier to figure out how to distribute something

### Possible Drawbacks

None

### Verification Process

1. Checkout this PR
2. Test the various scenarios:
   - Edit a published item that hasn't been distributed yet, checking both normal mode and fullscreen mode
   - Edit a published item that has been distributed, checking both normal mode and fullscreen mode
   - Edit an item that hasn't been published yet, checking both normal mode and fullscreen mode
   - Edit an item that was received through distribution, checking both normal mode and fullscreen mode
3. Ensure the new Distributor panel shows properly in all scenarios and toggling to the admin bar works properly in fullscreen mode

### Checklist:


### Applicable Issues

#597
